colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +211,81 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+"""
+Global variable for the cached CMake version number.
+
+When valid, this will be a pkg_resources.extern.packaging.version.Version.
+It may also have a boolean value False when the CMake version could not be
+determined to avoid trying to determine it again.
+"""
+_cached_cmake_version = None
+
+
+def get_cmake_version():
+    """
+    Get the CMake version.
+
+    The function caches the result on the first invocation and reused that on
+    subsequent invocations.
+    The result is None when a version number could not be determined.
+
+    :returns: The version as reported by `CMAKE_EXECUTABLE --version`, or None
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    global _cached_cmake_version
+    if _cached_cmake_version is None:
+        cmake_version = _parse_cmake_version()
+        if cmake_version is not None:
+            _cached_cmake_version = cmake_version
+        else:
+            # flag to avoid retrying on subsequent calls
+            _cached_cmake_version = False
+
+    if _cached_cmake_version is False:
+        # in case the CMake version could not be determined
+        return None
```

Comment:
The logic around the cache value is pretty convoluted coming from the fact that the function uses a different "negative" value (`None`) than the cached value (`False`).

I aligned those in 6390dd148c23f910309afe802c76cbdc2dddff3d which gets rid of the majority of the logic in this function.